src/valkey-cli.c

```diff
@@ -2428,22 +2434,14 @@ static int cliSendCommand(int argc, char **argv, long repeat) {
                     /* Handle pubsub mode */
                     if (config.last_reply->elements >= 3) {
                         char *cmd = config.last_reply->element[0]->str;
-                        int count = config.last_reply->element[2]->integer;
-
-
-                        /* If it's an unsubscribe command, call the helper */
-                        if (strcmp(cmd, "unsubscribe") == 0 || strcmp(cmd, "punsubscribe") == 0 || strcmp(cmd, "sunsubscribe") == 0) {
-                            if (config.pubsub_mode && isPubsubPush(config.last_reply)) {
-                                processUnsubscribeReply(config.last_reply);
-                            }
-                        } 
-
-                        /* Handle subscribe commands */
-                        else if (strcmp(cmd, "subscribe") == 0 || strcmp(cmd, "psubscribe") == 0) {
-                            config.pubsub_unsharded_count = count;
-                        } else if (strcmp(cmd, "ssubscribe") == 0) {
-                            config.pubsub_sharded_count = count;
+                        
+                        /* Call the helper to handle both subscribe and unsubscribe commands */
+                        if (strcmp(cmd, "subscribe") == 0 || strcmp(cmd, "psubscribe") == 0 ||
+                            strcmp(cmd, "ssubscribe") == 0 || strcmp(cmd, "unsubscribe") == 0 ||
+                            strcmp(cmd, "punsubscribe") == 0 || strcmp(cmd, "sunsubscribe") == 0) {
+                            handlePubSubMode(config.last_reply);
                         }
```

Comment:
We can delete more. We don't need to check the command because we check it inside handlePubSubMode. If we call handlePubSubMode with some other reply, it does nothing.

```suggestion
                        handlePubSubMode(config.last_reply);
```

Also, we can delete the `if (config.last_reply->elements >= 3)` check above because that's also checked inside the helper.

Finally, can we delete other code to set `config.pubsub_mode = 1` a few lines below now?

```diff
                     if (num_expected_pubsub_push > 0 && !strcasecmp(config.last_reply->element[0]->str, command)) {
                         /* This pushed message confirms the
                          * [p|s][un]subscribe command. */
-                        if (is_subscribe && !config.pubsub_mode) {
-                            config.pubsub_mode = 1;
-                            cliRefreshPrompt();
-                        }
                         if (--num_expected_pubsub_push > 0) {
                             continue; /* We need more of these. */
                         }
```

Reply:
Oh I realized this earlier and had pushed those changes a couple minutes before you sent these comments. The new commits should contain these